modular/executor/execute_task.go

```diff
@@ -323,7 +333,13 @@ func (e *ExecuteModular) recoverByPrimarySP(ctx context.Context, task coretask.R
 		log.CtxErrorw(ctx, "EC recover data write piece fail", "pieceKey:", recoveryKey, "error", err)
 		return err
 	}
-
+	if task.BySuccessorSP() {
+		err = e.setPieceMetadata(ctx, task, pieceData)
+		if err != nil {
+			log.CtxErrorw(ctx, "failed to set piece meta data to DB", "objectName:", task.GetObjectInfo().GetObjectName(), "segment_idx", task.GetSegmentIdx(), "redundancy_idx", task.GetEcIdx(), "error", err)
```

Comment:
if the meta set failed ,  may need consider to delete the recovery data to make it a atomic action

Reply:
As the SP is still going to retry to acquire data, this would avoid the cost of repeatedly deleting and putting for same piece. If the SP no longer want to be the successor, this useless piece in piece store will be GC.